### PR TITLE
refactor: Fix near-stdx clippy warnings

### PR DESCRIPTION
**Pull request by pugachAG**
<details>
<summary>Warnings fixed</summary>

```
warning: this let-binding has unit value
  --> utils/stdx/src/lib.rs:11:5
   |
11 |     let () = AssertEqSum::<N, L, R>::OK;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: omit the `let` binding: `AssertEqSum::<N, L, R>::OK;`
   |
   = note: `#[warn(clippy::let_unit_value)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_unit_value

warning: this let-binding has unit value
  --> utils/stdx/src/lib.rs:90:5
   |
90 |     let () = AssertNonZero::<N>::OK;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: omit the `let` binding: `AssertNonZero::<N>::OK;`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_unit_value

warning: this returns a `Result<_, ()>`
   --> utils/stdx/src/lib.rs:102:1
    |
102 | pub fn as_chunks_exact<const N: usize, T>(slice: &[T]) -> Result<&[[T; N]], ()> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(clippy::result_unit_err)]` on by default
    = help: use a custom `Error` type instead
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#result_unit_err

warning: boolean to int conversion using if
   --> utils/stdx/src/lib.rs:127:25
    |
127 |     const OK: () = [()][if N == 0 { 1 } else { 0 }];
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace with from: `usize::from(N == 0)`
    |
    = note: `#[warn(clippy::bool_to_int_with_if)]` on by default
    = note: `(N == 0) as usize` or `(N == 0).into()` can also be valid options
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#bool_to_int_with_if
```
</details>

`let_unit_value` is suppressed since this is explicitly what we want in that case. Please note that `_ = AssertEqSum::<N, L, R>::OK;` produces the same clippy warning and having just `AssertEqSum::<N, L, R>::OK;` results in `path statement with no effect` rustc warning. Also we can't use `static_assertions` crate here since `near-stdx` should not have external dependencies as stated in `utils/stdx/Cargo.toml`.

This also includes refactor around static assertions to use std `assert!` which still works on compile time:

<details>
<summary>Assertion failure example</summary>

```
error[E0080]: evaluation of `AssertEqSum::<5, 1, 3>::OK` failed
   --> utils/stdx/src/lib.rs:125:20
    |
125 |     const OK: () = assert!(S == A + B);
    |                    ^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'assertion failed: S == A + B', utils/stdx/src/lib.rs:125:20
    |
    = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

note: the above error was encountered while instantiating `fn split_array::<5, 1, 3>`
  --> utils/stdx/src/lib.rs:61:36
   |
61 |     assert_eq!((&[0], &[2, 3, 4]), split_array(&[0, 1, 2, 3, 4]));
   |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0080`.
error: could not compile `near-stdx` due to previous error
```
</details>

Part of #8145

### CODE REVIEW

File: utils/stdx/src/lib.rs [hunk 1]
```diff
@@ -98,31 +102,41 @@ pub fn as_chunks<const N: usize, T>(slice: &[T]) -> (&[[T; N]], &[T]) {
...
+}
+
+const fn static_assert(ok: bool) {
+    // Array with a single unit element
+    let arr = [()];
+    // index == 0 ↔ ok == true
+    let index = !ok as usize;
+    // the following expression will fail with
+    // "index out of bounds" if ok is false
+    arr[index]
```
Comment (jakmeier) [inline utils/stdx/src/lib.rs:141]: Yeah that's still hacky but at least no longer obfuscated :D 

But I think we can do better! Apparently `const_panic` has been stable for a while, so a dependency to static_assertions is not even necessary. That is, you can just use regular `assert!` in a const fn. Example taken from https://internals.rust-lang.org/t/nicer-static-assertions/15986

```rust
const _: () = assert!(FOO == BAR, "assert_foo_equals_bar");
```

Comment (pugachAG) [reply utils/stdx/src/lib.rs]: that is awesome, thanks, updated!

File: utils/stdx/src/lib.rs [hunk 2]
```diff
@@ -98,31 +102,31 @@ pub fn as_chunks<const N: usize, T>(slice: &[T]) -> (&[[T; N]], &[T]) {
     (head, tail)
 }
 
-/// Like `as_chunks` but returns an error if there’s a remainder.
-pub fn as_chunks_exact<const N: usize, T>(slice: &[T]) -> Result<&[[T; N]], ()> {
+/// Chunks if `as_chunks` has no remainder.
+pub fn as_chunks_exact<const N: usize, T>(slice: &[T]) -> Option<&[[T; N]]> {
```
Comment (nagisa) [inline utils/stdx/src/lib.rs:106]: I think we should keep using a `Result` here. It fits better semantically in that a failure here signals that the call is ill formed.

An error type like 

```rust
#[derive(Debug, Clone)]
struct InexactChunkingError(());
impl std::error::Error for InexactChunkingError {}
impl std::fmt::Display for InexactChunkingError { ... }
```

probably isn’t too onerous to add for this.

Comment (pugachAG) [reply utils/stdx/src/lib.rs]: makes sense, thanks for the suggestion

**Review by jakmeier — APPROVED** [review]

**Review by nagisa — APPROVED** [review]

**Pull request merged**